File: Universidad de Granada/Programación y Diseño Orientada a Objetos/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/dado.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#authors: Miguel Ángel Benítez Alguacil
#         Najib Saadouli Arco

require "singleton"
module ModeloQytetet
  class Dado
    include Singleton
    
    def initialize
      @valor = 0
    end
    
    attr_reader :valor
    
    def tirar
      @valor = rand(5)+1    #rand(5)+1
      return @valor
    end
    
    def to_s
      "Dado{ valor= #{@valor} }"
    end
    
  end
end

File: Universidad de Granada/Programación y Diseño Orientada a Objetos/QytetetJava/src/modeloqytetet/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;
import java.util.Random;
/**
 *
 * @author Miguel Ángel Benítez Alguacil
 *         Najib Saadouli Arco
 */

class Dado {
    Random random = new Random();
    private int valor;
    private static final Dado OBJETODADO = new Dado();
    
    private Dado(){
        valor = 0;
    }
    
    public static final Dado getDado(){
        return OBJETODADO;
    }
    
    public int getValor(){
        return valor;
    }
    
    int tirar(){
        valor = random.nextInt(6) + 1;
        return valor;
    }
}
